src/NuGet.Core/NuGet.Commands/VerifyCommand/VerifyCommandRunner.cs

```diff
@@ -58,10 +60,16 @@ public async Task<int> ExecuteCommandAsync(VerifyArgs verifyArgs)
                         _defaultFingerprintAlgorithm)).ToList();
 
                 var verifierSettings = SignedPackageVerifierSettings.GetVerifyCommandDefaultPolicy();
+                SettingSection trustedSignersSection = verifyArgs.Settings?.GetSection(TrustedSignersSectionName);
+                List<TrustedSignerItem> trustedSigners = trustedSignersSection?.Items.Select(c => c as TrustedSignerItem).Where(c => c != null).ToList();
+                IEnumerable<KeyValuePair<string, HashAlgorithmName>> allowUntrustedRootList = trustedSigners?
+                    .SelectMany(c => c.Certificates)
+                    .Where(c => c.AllowUntrustedRoot)
+                    .Select(c => new KeyValuePair<string, HashAlgorithmName>(c.Fingerprint, c.HashAlgorithm));
```

Comment:
[`nuget.exe verify command spec`](https://github.com/NuGet/Home/wiki/NuGet-Verify-Command#config-file-for-repository-signature-verification) specifies that `-ConfigFile  option from NuGet.Config file is used for only repository signature verification. `, which makes me think that `verify` command should not apply all the settings from the `NuGet.Config` file while verifying package signature.

I posted this comment in another issue but here are the default settings used by `Verify` command. I would check with @dtivel and @heng-liu just to make sure required options are read from the `NuGet.Config` file.

Setting | Verify Command default setting
-- | -- | 
AllowUnsigned | ❌
AllowIllegal | ❌
AllowUntrusted | ❌
AllowIgnoreTimestamp | ❌
AllowMultipleTimestamps | ✔️
AllowNoTimestamp | ✔️
AllowUnknownRevocation | ✔️
ReportUnknownRevocation |  ✔️
VerificationTarget | Targets unknown, Author and Repository signatures
SignaturePlacement | Primary, Counter signatures
RepositoryCountersignatureVerificationBehavior | IfExists (Verify a signature if it exists)
RevocationMode | Online unless "NUGET_CERT_REVOCATION_MODE" environment variable   is set

Reply:
Sure, let's wait for Damon to clarify for us. Mean time I `dotnet nuget verify` [spec](https://raw.githubusercontent.com/NuGet/Home/dev/proposed/archive/DotNetSignCommands.md). It does have `--config` file section, somehow it's not implemented.

`        --configfile                    The NuGet configuration file. If specified, only the settings from this file will be used. If not specified, the hierarchy of configuration files from the current directory will be used.`

Reply:
@JonDouglas 
Hi. I know it happened very long time ago. Could you be able to check if discrepancy between 2 specs? I couldn't see any other spec contributors from md file.

Reply:
@aortiz-msft 
Fyi: Should we ignore `--configFile` option? How about NuGet.config file current working directory?

Reply:
@kartheekp-ms 
Could you able to clarify your [comment about #10011](https://github.com/NuGet/Home/issues/10774#issuecomment-1005318333)? I'm bit confused now.  Shouldn't #10011 already be closed or at least P3 priority item based on your above?

Reply:
> This makes me think that verify command should not apply all the settings from the NuGet.Config file while verifying package signature.

What are the main concerns here? There should be a way to specify what configuration file is applied & if not, defaulting to the user location given it applies to all operations. If in a directory that contains a solution/project, they are overridden by solution level config settings.

Reply:
I didn't find anything in [nuget verify command spec](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-verify), but found a wiki (not sure if it's still up to date though): https://github.com/NuGet/Home/wiki/NuGet-Verify-Command#config-file-for-repository-signature-verification, saying that verify command will take the config file option.

Reply:
> @kartheekp-ms
Could you able to clarify your comment about #10011? I'm bit confused now. Shouldn't #10011 already be closed or at least P3 priority item based on your above comment since nuget.config file is ignored?

My message in that comment is, the behavior of nuget.exe verify and dotnet nuget verify command should match once we add an option to accept config file. The reason is we try to maintain parity between nuget.exe and dotnet commands. I agree Nikolche's comment (next to mine in the same issue) because the changes would impact both the commands because the code base is shared.

Reply:
> The reason is we try to maintain parity between nuget.exe and dotnet commands.

I see people in the NuGet team keep saying this, but the dotnet cli intentionally distanced itself from the older msbuild.exe, including changing defaults to make more sense given the decade(s) that passed since msbuild/nuget.exe was first created. Some examples are most commands changing default verbosity from normal to minimal, and changing from Microsoft/cmd.exe style `/arg` arguments to GNU style `-a` and `--arg` style. So, by design, the dotnet CLI is not 100% compatible with the older tools. A third example is that msbuild.exe does not do an implicit NuGet restore (the `-restore` argument), whereas the dotnet CLI does.

I agree with reading nuget.config files, and think it's a bug that nuget.exe verify didn't. But I don't like seeing "make dotnet nuget the same as nuget.exe" used as any justification since I think we can improve customer experience with the newer tool.